modules/picard/crosscheckfingerprints/main.nf

```diff
@@ -0,0 +1,62 @@
+process PICARD_CROSSCHECKFINGERPRINTS {
+    tag "$meta.id"
+    label 'process_medium'
+
+    conda (params.enable_conda ? "bioconda::picard=2.26.10" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/picard:2.26.10--hdfd78af_0' :
+        'quay.io/biocontainers/picard:2.26.10--hdfd78af_0' }"
+
+    input:
+    tuple val(meta), path(input1)
+    path(input2)
+    file haplotype_map
+
+    output:
+    tuple val(meta), path("*.crosscheck_metrics.txt")   ,emit: crosscheck_metrics
+    path "versions.yml"                                 ,emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+
+    def input1_string = input1.join(" --INPUT ")
```

Comment:
@matthdsm  i'm curious as to why `--INPUT` and `--SECOND_INPUT` are being joined to the inputs as to declaring them as params on L41 (actually wouldn't it be duplicating here!) and L42 ?

Reply:
Hi,
`input1` is a mandatory field for the tool,  so it makes sense (to me) to code that in the script part.
`input2` however is optional, so we need to need to make sure it exists. If it has one value the `join` won't do anything, so I add `--SECOND_INPUT` before it.

I'm still new to nextflow and groovy, so no idea if there's a better way to do this. All suggestions are welcome!